pkg/extension/sumologicextension/extension.go

```diff
@@ -337,6 +191,43 @@ func (se *SumologicExtension) sendHeartbeat() error {
 
 }
 
+// storeCollectorCredentials stores collector credentials in a file in directory
+// as specified in CollectorCredentialsPath. The credentials are encrypted using
+// hashed collector name.
+func (se *SumologicExtension) storeCollectorCredentials() error {
+	if err := ensureStoreCredentialsDir(se.conf.CollectorCredentialsPath); err != nil {
+		return err
+	}
+	filenameHash, err := hash(se.conf.CollectorName)
+	if err != nil {
+		return err
+	}
+	path := path.Join(se.conf.CollectorCredentialsPath, filenameHash)
+	collectorCreds, err := json.MarshalIndent(se.registrationInfo, "", " ")
+	if err != nil {
+		return err
+	}
+	encrypedCreds, err := encrypt(collectorCreds, se.conf.CollectorName)
+	if err != nil {
+		return err
+	}
+	return fmt.Errorf("failed to save credentials file '%s': %w",
+		path,
+		os.WriteFile(path, encrypedCreds, 0600),
+	)
+}
+
+// ensureStoreCredentialsDir checks if directory to store credentials exists,
+// if not try to create it.
+func ensureStoreCredentialsDir(path string) error {
```

Comment:
This doesn't really need to mention credentials dir: it just checks if a dir exists and if not it creates it.